lib/chibi/snow/utils.scm

```diff
@@ -1,3 +1,71 @@
+(define (make-memoized-process->sexp cmd)
+  (let ((cache #f))
+    (lambda ()
+      (unless cache
+        (set! cache (process->sexp cmd)))
+      cache)))
+
+(define (write-to-string x)
+  (call-with-output-string (lambda (out) (write x out))))
+
+(define known-implementations
+  `((chibi "chibi-scheme" (chibi-scheme -V) "0.7.3"
+           ,(make-memoized-process->sexp
```

Comment:
Why not just use `delay`?

Reply:
It didn't occur to me. I just didn't use delay/force often enough to remember.